wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/UIAlertController+ConversationGuestOptions.swift

```diff
@@ -74,4 +87,25 @@ extension UIAlertController {
                                                    confirmAction: confirmAction,
                                                    completion: completion)
     }
+
+    static func confirmControllerWithTwoOptions(
+        optionOne: String,
+        optionTwo: String,
+        completion: @escaping (Bool) -> Void
+    ) -> UIAlertController {
+        let controller = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+        let optionOneAction = UIAlertAction(title: optionOne, style: .default) { _ in
+            completion(true)
+        }
+
+        let optionTwoAction = UIAlertAction(title: optionTwo, style: .default) { _ in
+            completion(false)
```

Comment:
we return false for option2 and cancel, how do you differentiate the two then? Would cancelling then create the link without password?

Reply:
@netbe Nice catch. I updated the code for the cancel action and now we just dismiss the controller instead of setting the completion to false.

```swift
controller.addAction(.cancel {
            controller.dismiss(animated: true)
        })
```